src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/FindUnwantedIslandsTask.cpp

```diff
@@ -0,0 +1,136 @@
+ * MA 02110-1301, USA.
+ */
+
+#include "FindUnwantedIslandsTask.h"
+
+#include <U2Core/DNASequenceUtils.h>
+#include <U2Core/PrimerDimersFinder.h>
+#include <U2Core/PrimerStatistics.h>
+#include <U2Core/U2SafePoints.h>
+
+#include "utils/UnwantedConnectionsUtils.h"
+
+namespace U2 {
+
+FindUnwantedIslandsTask::FindUnwantedIslandsTask(const U2Region& _searchArea, int _possibleOverlap, const QByteArray& _sequence, bool _isComplement)
+    : Task("Find Unwanted Islands Task", TaskFlags_FOSCOE),
+      searchArea(_searchArea),
+      possibleOverlap(_possibleOverlap),
+      sequence(_sequence),
+      isComplement(_isComplement) {}
+
+void FindUnwantedIslandsTask::run() {
+    taskLog.details(tr("Searching of unwanted islands and areas between them "
+                       "in the region \"%1\" (+ %2 nucleotides to the %3, deep into the amplified fragment) has been started")
+                       .arg(regionToString(searchArea)).arg(possibleOverlap).arg(isComplement ? "left" : "right"));
+    taskLog.details(tr("The following unwanted parametes are used. Free Gibbs energy: %1 kj/mol, "
+                       "melting temperature: %2 C, maximum dimer length: %3 nt")
+                       .arg(UNWANTED_DELTA_G).arg(UNWANTED_MELTING_TEMPERATURE).arg(UNWANTED_MAX_LENGTH));
+    /**
+     * Index of the left nucleotide in the searching area.
+     */
+    int leftNucleotide = searchArea.startPos;
+    /**
+     * Index of the right nucleotide in the searching area.
+     * @settings.overlapLength.maxValue is the area extending deep into the amplified fragment.
+     */
+    const int rightNucleotide = searchArea.endPos() + possibleOverlap;
+    auto text2LogAboutFoundRegion = [this](const U2Region& reg) {
+        taskLog.details(tr("The region between unwanted islands has been found: %1").arg(regionToString(reg)));
+    };
+
+    int lengthBetweenIslands = 0;
+    int startNucleotideNumber = leftNucleotide;
+    regionsBetweenIslands.clear();
+    stateInfo.setProgress(0);
+    int progressZeroPoint = startNucleotideNumber;
+    while (leftNucleotide /*+ ISLAND_LENGTH*/ <= rightNucleotide) {
+        U2Region islandCandidate(leftNucleotide, ISLAND_LENGTH);
+        bool isIsland = hasUnwantedConnections(islandCandidate);
+        if (isIsland) {
+            //The obvious limit - we don't need regions which couldn't fit the primer
+            if (lengthBetweenIslands != 0/*>= overlap.minValue*/) {
+                U2Region newRegion(startNucleotideNumber, lengthBetweenIslands);
+                text2LogAboutFoundRegion(newRegion);
+                regionsBetweenIslands << newRegion;
+            }
+            startNucleotideNumber = leftNucleotide;
+            lengthBetweenIslands = 0;
+        } else {
+            lengthBetweenIslands++;
+        }
+        leftNucleotide++;
+        stateInfo.setProgress(100 - (double(rightNucleotide - leftNucleotide) / double(rightNucleotide - progressZeroPoint)) * 100);
```

Comment:
According to the code above progressZeroPoint may be equal to rightNucleotide -> division by zero

Reply:
Well actually, it's impossible, cuz the region length is 1 a least. But I'll fix this, just in case